crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -133,6 +155,236 @@ fn is_fast_corner(
     false
 }
 
+/// Calculate the FAST corner score for a pixel using Sum of Absolute Differences (SAD).
+/// Based on https://www.edwardrosten.com/work/rosten_2006_machine.pdf.
+/// Returns a tuple of (is_corner: bool, score: i32)
+fn get_fast_corner_score(
+    src: &[u8],
+    pixel_idx: i32,
+    offsets: [i32; 16],
+    threshold: u8,
+    arc_length: u8,
+) -> (bool, i32) {
+    let center_pixel = unsafe { *src.get_unchecked(pixel_idx as usize) };
+    let lower_threshold = &center_pixel.saturating_sub(threshold);
+    let upper_threshold = &center_pixel.saturating_add(threshold);
+
+    // Helper to get pixel value efficiently with unchecked access
+    let get_pixel_from_offset =
+        |off_idx: usize| unsafe { src.get_unchecked((pixel_idx + offsets[off_idx]) as usize) };
+
+    // Fast rejection test - check if at least 3 of the 4 high-speed test points are different enough
+    let p1 = get_pixel_from_offset(0);
+    let p5 = get_pixel_from_offset(4);
+    let p9 = get_pixel_from_offset(8);
+    let p13 = get_pixel_from_offset(12);
+
+    let m0 = (p1 >= lower_threshold) as u8 + 
```

Comment:
In the reference implementation of OpenCV they were checking with Boolean masks. Maybe the casting is adding overhead